src/client/components/formComponent/formComponent.js

```diff
@@ -1,32 +1,42 @@
 import React from 'react';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import RenderFormFields from '../renderFormFields/renderFormFields';
 import Button from '../button';
+import Spinner from '../../assets/images/spinner.gif';
 import './formComponent.css';
 
-const FormComponent = ({
-  fieldsToRender,
-  onInputChange,
-  onFormSubmit,
-  userData,
-}) => (
-  <form onSubmit={onFormSubmit} className="form-body" autoComplete="off">
-    <RenderFormFields
-      fieldsToRender={fieldsToRender}
-      onInputChange={onInputChange}
-      userData={userData}
-    />
-    <div className="field">
-      <Button className="submit" value="Submit" type="submit" />
-    </div>
-  </form>
-);
+const FormComponent = (props) => {
+  let value = 'Submit';
+
+  if (props.loading) {
+    value = <img src={Spinner} alt="Authentication spinner" />;
```

Comment:
I would suggest to use better approach then inserting html tag in button’s value

Reply:
worked out